packages/dcos-integration-test/extra/test_metrics.py

```diff
@@ -336,6 +336,14 @@ def test_metrics_containers_nan(dcos_api_session):
         assert metric_value == '', 'unexpected metric value: {}'.format(metric_value)
 
 
+@retrying.retry(wait_fixed=(2 * 1000), stop_max_delay=(150 * 1000))
+def test_zookeeper_metrics_collected(dcos_api_session):
+    for master in dcos_api_session.masters:
+        response = dcos_api_session.get('/metrics', scheme='http', host=master, port=61091)
+        assert response.status_code == 200, 'Status code: {}'.format(response.status_code)
+        assert 'zookeeper_' in response.text
```

Comment:
Is this enough to tell us what we want to know?

Reply:
The output of `/metrics` must be massive, wow. We didn't specify any filter criterion and simply get everything? :) I should have a look at that API to better understand this.

Reply:
@adamtheturtle All ZooKeeper metrics start with this string. I could specify a particular one that we expect but I don't want the test to break if different metrics are outputted or the old ones are renamed.
@jgehrcke It's pretty neat how this works:
https://github.com/dcos/dcos/blob/53a31071349cd0cf6560ef48e5bbd36489e97022/gen/dcos-config.yaml#L1254
The reponse will no grow beyond a specified size, but that's why we need to retry the request like a Prometheus client to not miss the metrics being outputted.